docs/src/pages/components/masonry/masonry.md

```diff
@@ -0,0 +1,55 @@
+---
+title: React Masonry component
+components: Masonry, MasonryItem
+githubLabel: 'component: Masonry'
+---
+
+# Masonry
+
+<p class="description">Masonry lays out contents of different sizes as blocks of the same width and variable height with configurable gaps.</p>
+
+Masonry maintains a list of content blocks with a consistent width but variable height. The contents are ordered by row. If a row is already filled with the specified number of columns, the next item starts another row, and it is added to the shortest column.
+
+{{"component": "modules/components/ComponentLinkHeader.js", "design": true}}
+
+## Basic Masonry
+
+`<Masonry />` is a container for one or more `<MasonryItem />`s. `<MasonryItem />` can receive any element including `<div />` and `<img />`. Also, it is important to note that each `<MasonryItem />` accepts only one element.
+
+{{"demo": "pages/components/masonry/BasicMasonry.js", "bg": true}}
+
+## Image Masonry
+
+{{"demo": "pages/components/masonry/ImageMasonry.js", "bg": true}}
+
+## Columns
+
+By passing `columns` to `<Masonry />`, you can configure the number of columns of your masonry.
+
+{{"demo": "pages/components/masonry/FixedColumns.js", "bg": true}}
+
+`columns` accepts responsive values:
+
+{{"demo": "pages/components/masonry/ResponsiveColumns.js", "bg": true}}
+
+## Spacing
+
+By passing `spacing` to `<Masonry />`, you can configure the spacing between `<MasonryItem />`s. It is important to note that `spacing` is a factor of the theme's spacing.
+
+{{"demo": "pages/components/masonry/FixedSpacing.js", "bg": true}}
+
+`spacing` accepts responsive values:
+
+{{"demo": "pages/components/masonry/ResponsiveSpacing.js", "bg": true}}
+
+## Column spanning
+
+By passing `columnSpan` to `<MasonryItem />`, you can configure the number of columns taken up by each `<MasonryItem />`.
```

Comment:
Best to be honest in the description about the obvious limitation of this approach: 
![image](https://user-images.githubusercontent.com/357702/127913711-7280f1f8-9df4-4ce9-bfaf-0a7435bd6d3b.png)

(Not sure why the right border gets cut off in Codesandbox)

Reply:
I agree. I explained its limitation with a demo. The border cut-off is appearing on codesandbox because html's `box-sizing` is not set to `border-box` while MUI demo page has.

Reply:
I resolved it in this commit: https://github.com/mui-org/material-ui/pull/27439/commits/ac7bf9dc9ea8dbfb60dbacf96a63a99427ff00b1